src/imp/sp1.rs

```diff
@@ -101,70 +42,30 @@ pub(crate) fn init_state_with_cap_and_msg<'a>(
     cap: &Fr,
     msg: &[Fr],
 ) -> &'a mut State {
-    const ZERO_TWO: [Fr; 2] = [Fr::zero(), Fr::zero()];
+    static ZERO_TWO: [Fr; 2] = [Fr::zero(), Fr::zero()];
 
-    match msg.len() {
-        0 => unsafe {
-            asm!(
-                "ecall",
-                in("t0") MEMCPY_32,
-                in("a0") cap,
-                in("a1") (state.as_mut_ptr() as *mut Fr),
-            );
-            asm!(
-                "ecall",
-                in("t0") MEMCPY_64,
-                in("a0") &ZERO_TWO,
```

Comment:
actually this is wrong, &ZERO_TWO is a fat pointer

Reply:
In `sp1-intrinsics`, the wrapping require the ptr be `thin`, we can avoid this kind of issues.